cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -448,6 +445,92 @@ void build_host(raft::resources const& handle,
                 raft::host_matrix_view<const uint8_t, int64_t, raft::row_major> dataset,
                 cuvs::neighbors::cagra::index<uint8_t, uint32_t>& idx);
 
+/**
+ * @defgroup cagra_cpp_index_add_nodes CAGRA graph node addition functions
+ * @{
+ */
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::device_mdspan<const float, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<float, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::host_mdspan<const float, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<float, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::device_mdspan<const int8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<int8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::host_mdspan<const int8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<int8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::device_mdspan<const uint8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<uint8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::host_mdspan<const uint8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<uint8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+/**
+ * @defgroup cagra_cpp_index_extend CAGRA extend functions
+ * @{
+ */
+void extend(raft::resources const& handle,
```

Comment:
These all need to be documented with descriptions and copy/paste usage examples. Please also document all of them so that any function that shows up in the API docs will contain the proper documentation.